examples/powertools-examples-core/sam/src/main/java/helloworld/AppStream.java

```diff
@@ -17,22 +17,40 @@
 import com.amazonaws.services.lambda.runtime.Context;
 import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
 import com.fasterxml.jackson.databind.ObjectMapper;
+
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
-import java.util.Map;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import software.amazon.lambda.powertools.logging.Logging;
 import software.amazon.lambda.powertools.metrics.Metrics;
 
+import java.io.InputStreamReader;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+
 public class AppStream implements RequestStreamHandler {
     private static final ObjectMapper mapper = new ObjectMapper();
+    private final static Logger log = LogManager.getLogger(AppStream.class);
 
     @Override
     @Logging(logEvent = true)
     @Metrics(namespace = "ServerlessAirline", service = "payment", captureColdStart = true)
-    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
-        Map map = mapper.readValue(input, Map.class);
+    public void handleRequest(InputStream input, OutputStream output, Context context) {
```

Comment:
I think it would be helpful to add some info here about why you would use `RequestStreamHandler` - e.g. to make it clear that it has nothing to do with HTTP streaming.

Reply:
Note added to explain why RequestStreamHandler is preferred here

Reply:
I think it's not quite right - I think the use case is more typically "we need to control [de]serialization ourselves".

 I find "rather than a simple HTTP request-response model" a little confusing, because you still have exactly the same surrounding HTTP interaction as you would with a 'normal' RequestHandler, you're just choosing to read-and-write the `Body` using java streams rather than letting Lambda handle the serialization for you.

@jeromevdl I don't know if you have some super clear wording handy? The core Lambda documentation isn't that prescriptive.

Reply:
I agree wording is not clear or even wrong. `RequestStreamHandler` is not linked to [Lambda streaming](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-response-streaming/) which was announced recently. It is used when you want to use your own serialization: https://docs.aws.amazon.com/lambda/latest/dg/java-handler.html#java-handler-interfaces. I agree the name of the interface is not super well chosen but it has nothing to do with streams...